### PR TITLE
Add test for duplicate module procedure names

### PR DESCRIPTION
**Pull request by mohitdhingra22**
I have added a test case to check for duplicate module procedure names and registered it in tests.toml.

### CODE REVIEW

**Comment by gxyd** [comment]
Please link the issue that this PR fixes in your PR description. Mention something like, fixes #xxxx , where xxxx is the issue number, GitHub will automatically link it to that issue.

**Comment by assem2002** [comment]
It should be closing this one https://github.com/lfortran/lfortran/issues/4222 right?

**Comment by gxyd** [comment]
> It should be closing this one #4222 right?

Yes, I think that's the one.

File: tests/tests.toml [hunk 1]
```diff
@@ -4138,3 +4138,7 @@ run = true
 [[test]]
 filename = "exit2.f90"
 run = true
+
+[[test]]
+name = "test_duplicate_module_procedures"
+path = "errors/test_duplicate_module_procedures.f90"
```
Comment (assem2002) [inline tests/tests.toml:4144]: Please change that to 
```suggestion

[[test]]
filename = "errors/test_duplicate_module_procedures.f90"
run = true
```
then run `run_tests.py  -u`, then push that commit.

**Comment by gxyd** [comment]
This PR I think mistakenly adds a binary file as well, i.e. `lfortran` itself. No need to commit that, be careful, as it's a dangerous thing, I've done in one of the PR's myself.

File: tests/tests.toml (hunk 1 above)
Comment (assem2002) [reply tests/tests.toml:4144]: with correct fix it should raise an error like this one ` Entity ‘test_01’ at (1) is already present in the interface`

**Comment by assem2002** [comment]
That's good to notice, I haven't notice that :).

**Review by Pranavchiku — COMMENTED** [review]
If I understand correctly, we are registering a test that should throw an error but it is not throwing an error in `main` or in this PR. If that is the case, I would suggest not to add it, rather add the test in the same PR which fixes ( throws error for ) it.

**Review by kmr-srbh — CHANGES_REQUESTED** [review]
@mohitdhingra22 thanks for this PR! I think we need to report an error for this. Please see `add_generic_procedures()` function in the `ast_symboltable_visitor.cpp` file. We need to catch duplicate symbols when pushing inside `Vec<ASR::symbol_t*> symbols` and report the error.

The added test is fine. Please make the above change and we will review it again.

File: tests/reference/run_dbg-runtime_stacktrace_01-dcc746e.stderr [hunk 2]
```diff
@@ -1,9 +1 @@
-code generation error: asr_to_llvm: module failed verification. Error:
-!dbg attachment points at wrong subprogram for function
-!3 = distinct !DISubprogram(name: "expr2", scope: !1, file: !1, line: 1, type: !4, flags: DIFlagPrototyped, spFlags: DISPFlagDefinition, unit: !0, retainedNodes: !2)
-i32 (i32, i8**)* @main
-  call void @_lpython_call_initial_functions(i32 %0, i8** %1), !dbg !7
-!7 = !DILocation(line: 1, column: 1, scope: !8)
-!8 = distinct !DISubprogram(name: "main", scope: !1, file: !1, line: 7, type: !4, flags: DIFlagPrototyped, spFlags: DISPFlagDefinition, unit: !0, retainedNodes: !2)
-!8 = distinct !DISubprogram(name: "main", scope: !1, file: !1, line: 7, type: !4, flags: DIFlagPrototyped, spFlags: DISPFlagDefinition, unit: !0, retainedNodes: !2)
-
+semantic error: The `runtime stacktrace` is not enabled. To get the stack traces or debugging information, please re-build LFortran with `-DWITH_RUNTIME_STACKTRACE=yes`
```
Comment (kmr-srbh) [inline tests/reference/run_dbg-runtime_stacktrace_01-dcc746e.stderr:1]: This change is incorrect. In fact the previous change was incorrect too. It seems to have come from 0d03807. The current output from `main` is incorrect too. We used to have a stacktrace here in the `stderr` output. @advikkabra could you please have a look into it?
